File: API/PageAndRDLX/FontResolver/C#/FontResolver/BarcodeFontResolver.cs
using GrapeCity.ActiveReports;
using GrapeCity.Documents.Text;
using System;

namespace ActiveReports.Samples.FontResolver
{
	internal sealed class BarcodeFontResolver : IFontResolver
	{
		public static IFontResolver Instance = new BarcodeFontResolver();

		FontCollection _fonts;

		private BarcodeFontResolver()
		{
			_fonts = new FontCollection();
			// load Windows fonts
			_fonts.RegisterDirectory(Environment.GetFolderPath(Environment.SpecialFolder.Fonts));
			// load barcode fonts
			_fonts.RegisterDirectory("../../../../../../Fonts");
			_fonts.DefaultFont = _fonts.FindFamilyName("Arial");
		}

		FontCollection IFontResolver.GetFonts(string familyName, bool isBold, bool isItalic)
		{
			var fonts = new FontCollection();
			fonts.Add(_fonts.FindFamilyName(familyName, isBold, isItalic) ?? _fonts.DefaultFont);
			return fonts;
		}
	}
}

File: API/PageAndRDLX/FontResolver/VB.NET/FontResolver/BarcodeFontResolver.vb
Imports GrapeCity.ActiveReports
Imports GrapeCity.Documents.Text

Friend NotInheritable Class BarcodeFontResolver
    Implements IFontResolver

    Public Shared Instance As IFontResolver = New BarcodeFontResolver()
    Private _fonts As FontCollection

    Private Sub New()
        _fonts = New FontCollection()
        _fonts.RegisterDirectory(Environment.GetFolderPath(Environment.SpecialFolder.Fonts))
        _fonts.RegisterDirectory("../../../../../../Fonts")
        _fonts.DefaultFont = _fonts.FindFamilyName("Arial")
    End Sub

    Private Function IFontResolver_GetFonts(familyName As String, isBold As Boolean, isItalic As Boolean) As FontCollection Implements IFontResolver.GetFonts
        Dim fonts = New FontCollection()
        fonts.Add(If(_fonts.FindFamilyName(familyName, isBold, isItalic), _fonts.DefaultFont))
        Return fonts
    End Function
End Class
